### ISSUE TITLE
Certificate Verify Failed on Amazon Linux 2 

### ISSUE THREAD

**Issue by geowa4** [osquery/osquery#7691]
# Bug report

When launching Osquery on Amazon Linux 2, queries hitting the EC2 API (or any AWS API) fail with a "certificate verify failed" message. I also noticed an issue sending result logs to Kinesis.

```
osquery> SELECT instance_id AS ec2_instance_id FROM ec2_instance_tags LIMIT 1;
E0715 13:21:47.181819 22291 aws_util.cpp:223] Exception making HTTP POST request to URL (https://ec2.us-east-1.amazonaws.com): certificate verify failed
```

This is my yum source for Osquery.

```
      name = osquery
   baseurl = https://s3.amazonaws.com/osquery-packages/rpm/$basearch/
mirrorlist =
   enabled = 1
  gpgcheck = 1
    gpgkey = https://pkg.osquery.io/rpm/GPG
```

And what it installed.

```
         name = osquery
      version = 5.3.0
      release = 1.linux
       source = osquery-5.3.0-1.linux.src.rpm
         size = 52373069
         sha1 = 67bf39cdb6151d5bbcf9243eb0e354299208c888
         arch = x86_64
        epoch =
 install_time = 1657759087
       vendor = osquery
package_group = default
```

Since this is a certificate issue, I checked that ca-certificates is also installed.

```
         name = ca-certificates
      version = 2021.2.50
      release = 72.amzn2.0.3
       source = ca-certificates-2021.2.50-72.amzn2.0.3.src.rpm
         size = 901298
         sha1 = 60b73ef29e9073c2ec9e04e6c6a70d17e01ec773
         arch = noarch
        epoch =
 install_time = 1655146360
       vendor = Amazon Linux
package_group = System Environment/Base
``` 

And the certificate information for ec2.us-east-1.amazonaws.com.

```
   subject_key_identifier = 2A:18:83:72:1D:D5:AD:FA:33:52:7B:82:19:35:CC:6A:B4:43:3B:BF
 authority_key_identifier = keyid:59:A4:66:06:52:A0:7B:95:92:3C:A3:94:07:27:96:74:5B:F9:3D:D0
                key_usage = Digital Signature, Key Encipherment
       extended_key_usage = TLS Web Server Authentication, TLS Web Client Authentication
                 policies = Policy: 2.23.140.1.2.1
subject_alternative_names = DNS:ec2.us-east-1.amazonaws.com, DNS:ec2.amazonaws.com, DNS:us-east-1.ec2.amazonaws.com, DNS:*.ec2.us-east-1.vpce.amazonaws.com, DNS:api.ec2.us-east-1.aws, DNS:ec2.us-east-1.api.aws
 issuer_alternative_names =
              info_access = OCSP - URI:http://ocsp.sca1b.amazontrust.com;CA Issuers - URI:http://crt.sca1b.amazontrust.com/sca1b.crt
      subject_info_access =
          policy_mappings =
              has_expired = 0
         basic_constraint = CA:FALSE
         name_constraints =
       policy_constraints =
                      pem =
```

### What operating system and version are you using?

`osqueryi --line "SELECT version, build, platform FROM os_version;"`

```
 version = 2
   build =
platform = amzn
```

### What version of osquery are you using?

`osqueryi --line "SELECT version from osquery_info;"`

```
version = 5.3.0
```

### What steps did you take to reproduce the issue?

Launch Osquery on Amazon Linux 2.

### What did you expect to see?

EC2 instance metadata

### What did you see instead?

Error message

**Comment by Smjert** [comment]
@Hello @geowa4 this is a known issue that has been fixed in 5.4.0: https://github.com/osquery/osquery/pull/7652

**Comment by geowa4** [comment]
@Smjert Thank youI I don't know how I missed that issue.
